### PR TITLE
Remove use of patchelf for ROCm wheels

### PR DESCRIPTION
**Pull request by suffiank**
The _patchelf_ utility has a bug that corrupts the binary and is now manifesting on import torch_ort.
```
>>> from torch_ort import ORTModule
Inconsistency detected by ld.so: dl-version.c: 205: _dl_check_map_versions: Assertion `needed != NULL' failed!
```
This makes the wheel unusable and is therefore a blocker.

Will review strategy in future as not using _patchelf_ ships AMD libraries with binary.
```
>> ls /opt/conda/lib/python3.6/site-packages/onnxruntime_training.libs/
libMIOpen-25fdb483.so.1.0.40200 
libelf-0-3d9de92a.176.so 
libmpi-47bf9666.so.40.20.4 
libopen-rte-9ec95e4c.so.40.20.4 
libtinfo-10270e32.so.5.9
libamd_comgr-85ab0c71.so.2.0.40200 
libhsa-runtime64-3795dc66.so.1.3.40200 
libnuma-428e8baf.so.1.0.0 
librccl-d304d461.so.1.0.40200 
libz-745e0a09.so.1.2.7
libamdhip64-46b24d76.so.4.2.40200 
libhsakmt-888cff18.so.1.0.40200 
libopen-pal-af589381.so.40.20.4 
librocblas-a6d1559e.so.0.1.40200
```

The wheel size grows from 11 MB to 95 MB.

_edit: Removed the auditwheel step because shipping separate AMD libraries leads to conflict with torch linked libraries. This means the AMD libraries will NOT get shipped. And I removed the license notices._

### CODE REVIEW

**Comment by snnn** [comment]
Does this change impact the other EPs too?

**Comment by suffiank** [comment]
> Does this change impact the other EPs too?

No, because all libraries in the dependency list were associated with AMD ROCm.

**Comment by snnn** [comment]
What's the license of these file:
```
libMIOpen
libelf
libmpi
libopen-rte
libtinfo
libamd_comgr
libhsa-runtime64
libnuma
librccl
libz
libamdhip64
libhsakmt
libopen-pal
librocblas
libopen
librocblas
```

**Comment by suffiank** [comment]
> If we put them in the wheel, it is considered as a "distribution" of the libraries. So we need to examine if any of them has a strong copyleft or weak copyleft license like GPL or LGPL. All of the libs above must be registered to https://github.com/microsoft/onnxruntime/tree/master/cgmanifests

It should all be open source. But I will need to track down the licenses.

File: tools/ci_build/github/linux/copy_strip_binary.sh [hunk 1]
```diff
@@ -53,6 +53,7 @@ cp $SOURCE_DIR/README.md $BINARY_DIR/$ARTIFACT_NAME/README.md
 cp $SOURCE_DIR/docs/Privacy.md $BINARY_DIR/$ARTIFACT_NAME/Privacy.md
 cp $SOURCE_DIR/LICENSE $BINARY_DIR/$ARTIFACT_NAME/LICENSE
 cp $SOURCE_DIR/ThirdPartyNotices.txt $BINARY_DIR/$ARTIFACT_NAME/ThirdPartyNotices.txt
+cp $SOURCE_DIR/tools/ci_build/github/azure-pipelines/licenses/ROCmNotices.txt $BINARY_DIR/$ARTIFACT_NAME/ROCmNotices.txt
```
Comment (jywu-msft) [inline tools/ci_build/github/linux/copy_strip_binary.sh:56]: is this supposed to be done for non-ROCm packages?

Comment (suffiank) [reply tools/ci_build/github/linux/copy_strip_binary.sh]: I was having trouble propagating ROCmNotices.txt and I was trying to mimic Privacy.md to figure it out. It seems this script is not invoked during training package build. So I reverted it and awaiting validation.

**Comment by suffiank** [comment]
@snnn Can you review again? Thanks.

File: tools/ci_build/github/linux/copy_strip_binary.sh (hunk 1 above)
Comment (jywu-msft) [reply tools/ci_build/github/linux/copy_strip_binary.sh]: if it's for building a python package, it should be here: https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime_python.cmake#L303

Comment (jywu-msft) [reply tools/ci_build/github/linux/copy_strip_binary.sh]: nm, looks like you found it :)

**Comment by suffiank** [comment]
Actually, good chance this PR will be dropped due to conflicts between the HSA library loaded by torch and the one shipped by the auditwheel step. The driver cannot support two copies of the HSA runtime in a single process.

**Comment by guikarist** [comment]
Hi, I've came cross with this error:

```python
import onnxruntime
import numpy as np

ortvalue = onnxruntime.OrtValue.ortvalue_from_numpy(np.array([1, 2, 3]), 'cuda', 0)
```

Output:
```shell
Inconsistency detected by ld.so: dl-version.c: 205: _dl_check_map_versions: Assertion `needed != NULL' failed!
```

Is it related to this PR?

**Comment by snnn** [comment]
The error was caused by manylinux patchelf tool. If we don't use the tool, we won't have the error. 
If you can build onnxruntime from source, then you can forget manylinux and see goodbye to this error.

**Comment by snnn** [comment]
BTW, the error doesn't affect use.  It is only a problem when you want to use ldd.

**Pull request closed without merging**